glancesync/glancesync.py

```diff
@@ -439,10 +439,11 @@ def __upload_image(self, master_image, images_dict, regionobj):
         # filter properties to upload
         metadata_set = regionobj.target['metadata_set']
         properties = list(new_image.user_properties.keys())
-        for p in properties:
-            if p not in metadata_set and \
-                    p != 'kernel_id' and p != 'ramdisk_id':
-                del new_image.user_properties[p]
+        if len(metadata_set) > 0:
+            for p in properties:
+                if p not in metadata_set and \
+                        p != 'kernel_id' and p != 'ramdisk_id':
+                    del new_image.user_properties[p]
```

Comment:
You could define ``metadata_set`` and ``properties`` as ``set`` objects (instead of lists) and use the difference operator ``properties - metadata_set`` to get the elements if the former not present in the latter. And even better: you could also substract ``set(['kernel_id', 'ramdisk_id'])``.